compiler-rt/lib/tysan/tysan_interceptors.cpp

```diff
@@ -110,7 +110,7 @@ INTERCEPTOR(void *, malloc, uptr size) {
 
 #if SANITIZER_APPLE
 INTERCEPTOR(uptr, malloc_size, void *ptr) {
-  if (DlsymAlloc::Use())
+  if (DlsymAlloc::Use() || DlsymAlloc::PointerIsMine(ptr))
```

Comment:
Same question as for `realloc`: please document the state transitions we support.

Specifically: could it be that `DlSymAlloc::Use` is true but _NOT_ `DlsymAlloc::PointerIsMine(ptr)` (e.g. because the pointer was created before DlSymAlloc was enabled? That would be incorrect then I think.

Reply:
For `realloc`, during TySan init `realloc(0, size)` can make `DlSymAlloc::Use` is true but `DlsymAlloc::PointerIsMine(ptr)` is false.

Reply:
So do we need `DlsymAlloc::Use()` here? To me `DlsymAlloc::PointerIsMine(ptr)` sounds like what we actually care about?

Reply:
Yep that should do the trick, done ,thanks